src/optim/base.py

```diff
@@ -28,7 +46,7 @@ def train_base(model, opt, data, scheduler, iterations, acc_steps, batch_size, s
     t0 = time.time()
     while itr < iterations:
         for microstep_idx in range(acc_steps):  # gradient accumulation
-            x, y = get_batch(data['train'], sequence_length, batch_size, device=extra_args.device)
+            x, y = get_batch(data_train_iter, device=extra_args.device)
```

Comment:
I think this will fail as soon as we end one epoch as the iterator will run out of data and raise StopIteration. Similarly the problem can happen for eval.